### PR TITLE
Solutions for the second assignment

### PR DESCRIPTION
**Pull request by aliskin**

### CODE REVIEW

File: aliskin/assignment2/lca.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+def find_height(node):
+    current_node = node
+    height = 0
+    while current_node:
+        current_node = current_node.parent
+        height += 1
+    return height
+
+def find_lca(first_node, second_node):
```
Comment (ariava) [inline aliskin/assignment2/lca.py]: Can you please add comments to the function and explain your algorithm?

File: aliskin/assignment2/lca.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
+import unittest
+
+class TreeNode():
```
Comment (ariava) [inline aliskin/assignment2/lca.py]: Could you factor out the common class TreeNode and build_tree_from_preorder() in another file so you don't have to duplicate the code?

**Review by ariava — CHANGES_REQUESTED** [review]
Good job! Please have a look at my comments.

**Comment by aliskin** [comment]
Thanks for the feedback!
Arianna, I did what you asked. I renamed find_height function so probably the comment is not needed anymore.

**Review by ariava — APPROVED** [review]
Looks good! Good job!

**Pull request merged**